src/main/java/org/bigredbands/mb/views/PdfImage.java

```diff
@@ -1,60 +1,21 @@
 package org.bigredbands.mb.views;
 
-import java.awt.Color;
-import java.awt.Dimension;
-import java.awt.Graphics;
-import java.awt.Graphics2D;
-import java.awt.Shape;
 import java.util.HashMap;
-import java.util.HashSet;
-
-import javax.swing.JPanel;
 
+import org.bigredbands.mb.models.Field;
 import org.bigredbands.mb.models.RankPosition;
 
-public class PdfImage extends JPanel {
+public class PdfImage extends FieldView {
 
-    private float scaleFactor;
-    private Dimension containingDimension;
-    private MainView mainView;
     private HashMap<String, RankPosition> rankPositions;
 
-    public PdfImage(float scaleFactor, Dimension containingDimension, HashMap<String, RankPosition> rankPositions) {
-        super();
-
-        this.scaleFactor = scaleFactor;
-        this.containingDimension = containingDimension;
+    public PdfImage(Field field, HashMap<String, RankPosition> rankPositions) {
+        super(field);
         this.rankPositions = rankPositions;
-
     }
 
-    @Override
-    public void paintComponent(Graphics g) {
-        super.paintComponent(g);
-
-        Graphics2D g2d = (Graphics2D)g;
-        g2d.setColor(Color.WHITE);
-        g2d.fill(g2d.getClipBounds());
-
-        g.setColor(Color.BLACK);
-        FootballField.drawFieldLines(g, containingDimension, 0, 0, 0);
-        FootballField.drawHashes(g, (int) (FootballField.END_ZONE_LENGTH * scaleFactor), 0, scaleFactor);
-
-        FootballField.drawRanks(
-                FootballField.createShapes(rankPositions, (int) (FootballField.END_ZONE_LENGTH * scaleFactor), 0, scaleFactor),
-                new HashMap<String, Shape>(),
-                new HashSet<String>(),
-                g,
-                (int) (FootballField.END_ZONE_LENGTH * scaleFactor),
-                0,
-                scaleFactor);
-    }
-
-    public float getScaleFactor(){
-        return this.scaleFactor;
+    @Override 
+    public HashMap<String, RankPosition> getRankPositions() {
```

Comment:
Why not define this in the FieldView?

Reply:
Eventually, I plan to use this for the move thumbnails and the main editor as well. For `PdfImage`, getting the ranks is easy (they're just passed in from the `PDFGenerator`), but for the move thumbnail we'll need to call the controller to get the ranks for a specific move. The editor will need to get the ranks for the _current_ move, and possibly also deal with things like selected or transient ranks. So I went with the abstract method to allow each individual version of the view to define how it gets ranks.